File: 387/Solution.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        d = dict()
        for i in s:
            d[i] = d.get(i, 0)+1
        
        for i in range(0, len(s)):
            if d[s[i]] == 1:
                return i
        
        return -1
        

File: 387/Solution.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> x = new HashMap<Character, Integer>();
        
        for(int i=0; i<s.length();i++) {
            x.put(s.charAt(i), x.getOrDefault(s.charAt(i), 0)+1);
        }
        
        for(int i=0; i<s.length();i++) {
            if(x.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        
        return -1;
        
        
    }
}